packages/git-fetcher/test/index.ts

```diff
@@ -78,3 +78,25 @@ test('fetch a big repository', async () => {
     }, { manifest })
   await Promise.all(Object.values(filesIndex).map(({ writeResult }) => writeResult))
 })
+
+test('still able to shallow fetch for allowed hosts', async () => {
+  const cafsDir = tempy.directory()
+  const fetch = createFetcher().git
+  const manifest = pDefer<DependencyManifest>()
+  const { filesIndex } = await fetch(
+    createCafsStore(cafsDir),
+    {
+      commit: 'c9b30e71d704cd30fa71f2edd1ecc7dcc4985493',
+      repo: 'https://github.com/kevva/is-positive.git',
+      type: 'git',
+    },
+    {
+      manifest,
+      gitShallowHosts: ['github.com'],
+    }
+  )
+  expect(filesIndex['package.json']).toBeTruthy()
```

Comment:
I can't really differentiate in test whether it is fetched using full clone or shallow clone, unless the git repo queries some git history on their prepare script. Anyway this test show that when it's fetched using shallow clone, it can still be fetched successfully

Reply:
I guess you can mock execa and check how it was called.